app/models/line_item.rb

```diff
@@ -1,10 +1,12 @@
 class LineItem < ApplicationRecord
   belongs_to :order, optional: true
   belongs_to :product
-  belongs_to :cart, optional: true
-
-  validates :product_id, uniqueness: { scope: :cart_id }
-
+  belongs_to :cart, optional: true, counter_cache: true
```

Comment:
Also study how counter_cache finds column in other model

Also how can i change column name for counter cache

Reply:
Counter_cache finds the column like <model_name>_count in the table that it belongs to. 
If we want to specify the count column we can use the definition like `counter_cache: :name_of_column`